crates/transpiler/src/passes/instruction_duration_check.rs

```diff
@@ -0,0 +1,66 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_circuit::operations::Operation;
+use qiskit_circuit::operations::Param;
+
+/// Run duration validation passes.
+///
+/// Args:
+///     dag: DAG circuit to check instruction durations.
+///     acquire_align: Integer number representing the minimum time resolution to
+///         trigger acquisition instruction in units of dt.
+///     pulse_align: Integer number representing the minimum time resolution to
+///         trigger gate instruction in units of ``dt``.
+/// Returns:
+///     True if rescheduling is required, False otherwise.
+
+#[pyfunction]
+#[pyo3(signature=(dag, acquire_align, pulse_align))]
+pub fn run_instruction_duration_check(
+    dag: &DAGCircuit,
+    acquire_align: u32,
+    pulse_align: u32,
+) -> PyResult<bool> {
+    let num_stretches = dag.num_stretches();
+
+    //Rescheduling is not necessary
+    if (acquire_align == 1 && pulse_align == 1) || num_stretches != 0 {
+        return Ok(false);
+    }
+
+    //Check delay durations
```

Comment:
See https://github.com/Qiskit/qiskit/pull/14700/files#r2193260197

```suggestion
    // Check delay durations
```